packages/manager/cypress/e2e/core/oneClickApps/one-click-apps.spec.ts

```diff
@@ -0,0 +1,249 @@
+import { baseApps } from 'src/features/StackScripts/stackScriptUtils';
+import { stackScriptFactory } from 'src/factories/stackscripts';
+import { oneClickApps } from 'src/features/OneClickApps/oneClickApps';
+
+import type { StackScript } from '@linode/api-v4';
+import type { OCA } from '@src/features/OneClickApps/types';
+
+authenticate();
+
+describe('OneClick Apps (OCA)', () => {
+  before(() => {
+    cleanUp(['linodes']);
+  });
+
+  it('Lists all the OneClick Apps', () => {
+    interceptGetStackScripts().as('getStackScripts');
+
+    cy.visitWithLogin(`/linodes/create?type=One-Click`);
+
+    cy.wait('@getStackScripts').then((xhr) => {
+      const stackScripts: StackScript[] = xhr.response?.body.data ?? [];
+
+      const trimmedApps: StackScript[] = filterOneClickApps({
+        baseApps,
+        newApps: {},
+        queryResults: stackScripts,
+      });
+
+      // Check the content of the OCA listing
+      cy.findByTestId('one-click-apps-container').within(() => {
+        // Check that all sections are present (note: New apps can be empty so not asserting its presence)
+        cy.findByTestId('Popular apps').should('exist');
+        cy.findByTestId('All apps').should('exist');
+
+        trimmedApps.forEach((stackScript) => {
+          const { decodedLabel, label } = handleAppLabel(stackScript);
+
+          // Check that every OCA is listed with the correct label
+          cy.get(`[data-qa-select-card-heading="${label}"]`).should('exist');
+
+          // Check that every OCA has a drawer match
+          // This validates the regex in `mapStackScriptLabelToOCA`
+          // and ensures every app listed has a corresponding populated drawer
+          // This is only true for the apps defined in `oneClickApps.ts`
+          expect(
+            mapStackScriptLabelToOCA({
+              oneClickApps,
+              stackScriptLabel: decodedLabel,
+            })
+          ).to.not.be.undefined;
+        });
+      });
+
+      // Check drawer content for one OCA candidate
+      const candidate = trimmedApps[0].label;
+      const stackScriptCandidate = cy
+        .get(`[data-qa-selection-card-info="${candidate}"]`)
+        .first();
+      stackScriptCandidate.should('exist').click();
+
+      const app: OCA | undefined = mapStackScriptLabelToOCA({
+        oneClickApps,
+        stackScriptLabel: candidate,
+      });
+
+      ui.drawer
+        .findByTitle(trimmedApps[0].label)
+        .should('be.visible')
+        .within(() => {
+          containsVisible(app?.description);
+          containsVisible(app?.summary);
+          containsVisible(app?.website);
+        });
+      ui.drawerCloseButton.find().click();
+      ui.drawer.find().should('not.exist');
+
+      // Check the filtering of the apps
+      cy.scrollTo(0, 0);
+      const initialNumberOfApps = trimmedApps.length;
+      cy.findByPlaceholderText('Search for app name')
+        .should('exist')
+        .type(candidate);
```

Comment:
```suggestion
      const initialNumberOfApps = trimmedApps.length;
      cy.findByPlaceholderText('Search for app name')
        .scrollIntoView()
        .should('be.visible')
        .type(candidate);
```
This might be slightly clearer about what the test is trying to do but very minor difference